src/app/(nextjs_migration)/components/client/PageLoadEvent.tsx

```diff
@@ -7,16 +7,29 @@
 import { useEffect } from "react";
 import { usePathname } from "next/navigation";
 import { useGlean } from "../../../hooks/useGlean";
+import { useCookies } from "react-cookie";
+
+export type Props = {
+  userId: string;
+};
 
 // Empty component that records a page view on first load.
-export const PageLoadEvent = () => {
+export const PageLoadEvent = (props: Props) => {
+  const [cookies, setCookie] = useCookies(["userId"]);
+  const userId = props.userId;
+
+  // TODO decide if we want to do this here or on the server side
```

Comment:
If here, then presumably inside the `useEffect`, rather than on every render (which might happen very frequently).